profiling/src/profiling.rs

```diff
@@ -0,0 +1,806 @@
+use crate::bindings::{
+    datadog_php_profiling_get_profiling_context, zend_execute_data, zend_function, zend_string,
+    ZEND_INTERNAL_FUNCTION, ZEND_USER_FUNCTION,
+};
+use crate::{RequestLocals, Uri, PHP_VERSION};
+use crossbeam_channel::{Receiver, Sender, TrySendError};
+use ddprof::exporter::{File, Tag};
+use ddprof::profiles;
+use ddprof::profiles::api::{Function, Line, Location, Period, Sample};
+use log::{debug, error, info, trace, warn};
+use std::borrow::{Borrow, Cow};
+use std::collections::{HashMap, HashSet};
+use std::error::Error;
+use std::ffi::CStr;
+use std::hash::Hash;
+use std::mem::ManuallyDrop;
+use std::os::raw::c_char;
+use std::str::{FromStr, Utf8Error};
+use std::sync::atomic::{AtomicBool, AtomicU32, Ordering};
+use std::sync::{Arc, Mutex};
+use std::thread::JoinHandle;
+use std::time::{Duration, Instant, SystemTime};
+
+const UPLOAD_PERIOD: Duration = Duration::from_secs(67);
+
+// Guide: upload period / upload timeout should give about the order of
+// magnitude for the capacity.
+const UPLOAD_CHANNEL_CAPACITY: usize = 8;
+
+const WALL_TIME_PERIOD: Duration = Duration::from_millis(10);
+const WALL_TIME_PERIOD_TYPE: ValueType = ValueType {
+    r#type: Cow::Borrowed("wall-time"),
+    unit: Cow::Borrowed("nanoseconds"),
+};
+
+#[derive(Debug, Clone)]
+struct WallTime {
+    instant: Instant,
+    systemtime: SystemTime,
+}
+
+impl WallTime {
+    fn now() -> Self {
+        Self {
+            instant: Instant::now(),
+            systemtime: SystemTime::now(),
+        }
+    }
+}
+
+#[repr(C, align(16))]
+#[derive(Debug, Copy, Clone)]
+pub struct Uuid {
+    pub data: [u8; 16usize],
```

Comment:
The `usize` is superfluous here.
```suggestion
    pub data: [u8; 16],
```